### DISCUSSION TITLE
what is the best way to get workspaces to work

### DISCUSSION THREAD

**Discussion by MangoIV** [YaLTeR/niri#612] [Q&A] [answered]
Hi! I'm using waybar and I would love to have workspaces displayed in the bottom left. From the documentation in waybar, this should work with the use of the `wlr/workspaces` module. 

Is there anything special I have to do wrt `niri` to get this to work? 

Thanks in advance!

**Comment by YaLTeR** [comment, marked as answer]
Hi, the wlr workspaces protocol was in stasis in wayland-protocols merge requests last time I checked, and niri itself currently lacks a good IPC to implement a niri-specific module properly. That said, there's been some workarounds: https://github.com/YaLTeR/niri/discussions/356

**Reply by MangoIV** [reply]
thanks. I guess I'll wait until it's going to happen

**Reply by YaLTeR** [reply]
I finished the event stream IPC in niri, so you can use a git build of niri with my Waybar PR: https://github.com/Alexays/Waybar/pull/3551

**Reply by MangoIV** [reply]
Woah! Thank you so much ❤️

**Reply by MangoIV** [reply]
I have been trying this out and I have run into an issue, namely, the patched waybar outputs:
```
[2024-09-03 16:44:22.264] [error] Niri IPC: failed to start event stream
Fontconfig error: "/etc/fonts/fonts.conf", line 5: not well-formed (invalid token)
Fontconfig error: Cannot load config file from /etc/fonts/fonts.conf
basic_string::_M_create
```
I don't know what this could be caused by, my `/etc/fonts/fonts.conf` line 5 is: 
```
<include ignore_missing=yes>/etc/fonts/conf.d</include>
``` 
this doesn't look like it's incorrect - also, why would it fail to start the IPC if it cannot read that file? 

Thanks in advance!

**Reply by MangoIV** [reply]
ah no, my bad, I forgot to actually update `niri` as well :D

**Reply by YaLTeR** [reply]
The fontconfig error is unrelated. Make sure you have the latest git niri. Check if `niri msg event-stream` works.

**Reply by MangoIV** [reply]
yes, it all works well! there seem to be some glitches where workspaces appear that are not in use, but apart from that, all is well :)

**Reply by YaLTeR** [reply]
These are not glitches; there's always an empty workspace in the end.

**Reply by MangoIV** [reply]
![image](https://github.com/user-attachments/assets/96e7167c-3b59-4b09-8dc3-0e594e40f74a)
thank you!
